flang/include/flang/Optimizer/Builder/Character.h

```diff
@@ -1,4 +1,4 @@
-//===-- Lower/CharacterExpr.h -- lowering of characters ---------*- C++ -*-===//
+//===-------- CharacterExpr.h -- lowering of characters ---------*- C++ -*-===//
```

Comment:
I think the `----` (dash fill) should be at the after " characters" to be consistent with the LLVM style.

Reply:
Fixed this one as well as the headers in:

include/flang/Optimizer/Builder/BoxValue.h
include/flang/Optimizer/Builder/Character.h
include/flang/Optimizer/Builder/Complex.h
include/flang/Optimizer/Builder/FIRBuilder.h
include/flang/Optimizer/Builder/MutableBox.h
lib/Optimizer/Builder/MutableBox.cpp

Also fixed the file name in the header (CharacterExpr.h -> Character.h) here and where it was also wrong.